Modules/unicodedata.c

```diff
@@ -1457,28 +1433,45 @@ static struct PyModuleDef unicodedatamodule = {
 PyMODINIT_FUNC
 PyInit_unicodedata(void)
 {
-    PyObject *m, *v;
-
-    Py_SET_TYPE(&UCD_Type, &PyType_Type);
-
-    m = PyModule_Create(&unicodedatamodule);
-    if (!m)
+    PyObject *mod = PyModule_Create(&unicodedatamodule);
+    if (!mod) {
         return NULL;
+    }
 
-    PyModule_AddStringConstant(m, "unidata_version", UNIDATA_VERSION);
-    Py_INCREF(&UCD_Type);
-    PyModule_AddObject(m, "UCD", (PyObject*)&UCD_Type);
+    if (UCD_Type == NULL) {
+        UCD_Type = (PyTypeObject *)PyType_FromSpec(
+            &unicodedata_ucd_type_spec);
+        if (UCD_Type == NULL) {
+            return NULL;
```

Comment:
It seems like you want to refactor this function. In this case, please fix a memory leak: Py_DECREF(mod) is needed on error. Please add an "error:" label, and use "goto error".